### PR TITLE
[animation] animate routing point changes on edges

### PR DESCRIPTION
**Pull request by JanKoehnlein**

### CODE REVIEW

File: src/features/routing/manhattan-edge-router.ts [hunk 1]
```diff
@@ -280,6 +281,27 @@ export class ManhattanEdgeRouter extends LinearEdgeRouter {
         }
     }
 
+    /**
+     * Add artificial routing points to keep all angles rectilinear.
+     * 
+     * This makes edge morphing look a lot smoother, where RP positions are interpolated 
+     * linearly probably resulting in non-rectilinear angles. We don't add handles for 
+     * these additional RPs.
+     */
```
Comment (spoenemann) [inline src/features/routing/manhattan-edge-router.ts]: Linter warnings:
```
WARNING: /workspace/sprotty/src/features/routing/manhattan-edge-router.ts:286:7 - trailing whitespace
WARNING: /workspace/sprotty/src/features/routing/manhattan-edge-router.ts:287:89 - trailing whitespace
WARNING: /workspace/sprotty/src/features/routing/manhattan-edge-router.ts:288:87 - trailing whitespace
```

File: src/features/update/edge-morph-animation.ts [hunk 2]
```diff
@@ -0,0 +1,90 @@
+/********************************************************************************
+ * Copyright (c) 2019 TypeFox and others.
```
Comment (spoenemann) [inline src/features/update/edge-morph-animation.ts]: `2020` :-)

File: src/features/update/update-model.spec.ts [hunk 3]
```diff
@@ -342,4 +352,105 @@ describe('UpdateModelCommand', () => {
...
+            const edge = context.root.index.getById(edgeId) as SEdge;
+            expect(edge.routingPoints).to.be.eql([
+                { x: 64, y: 0 },
+                { x: 64, y: 128 }
+            ]);
+            animation1.tween(0.5, context);
+            expect(edge.routingPoints).to.be.eql([
+                { x: 100, y: 0 }, // midpoint between 64:0 and 136:0
+                { x: 100, y: 94 }  // midpoint between 64:128 and the interpolated 136:60
+            ]);
```
Comment (spoenemann) [inline src/features/update/update-model.spec.ts]: I get this output when running `yarn test` on Gitpod:
```
  1) UpdateModelCommand
       morphs edge:

      AssertionError: expected [ { x: 64, y: 0 }, { x: 64, y: 128 } ] to deeply equal [ { x: 100, y: 0 }, { x: 100, y: 94 } ]
      + expected - actual

       [
         {
      -    "x": 64
      +    "x": 100
           "y": 0
         }
         {
      -    "x": 64
      -    "y": 128
      +    "x": 100
      +    "y": 94
         }
       ]
      
      at Context.<anonymous> (src/features/update/update-model.spec.ts:377:46)
```

File: src/features/move/move.ts [hunk 4]
```diff
@@ -328,42 +317,64 @@ export class MorphEdgesAnimation extends Animation {
...
                     routingHandles: []
                 };
-                closestSnapshot.router.applySnapshot(memento.edge, newSnapshot);
+                closestSnapshot.router.applySnapshot(morph.memento.edge, newSnapshot);
             });
         }
         return this.model;
     }
+
+    protected growToSize(route: RoutedPoint[], targetSize: number): Point[] {
```
Comment (spoenemann) [inline src/features/move/move.ts]: This is very similar to `EdgeMorphAnimation.interpolateMissingPoints`. Could that be reused or extracted?

File: src/features/routing/manhattan-edge-router.ts (hunk 1 above)
Comment (JanKoehnlein) [reply src/features/routing/manhattan-edge-router.ts]: Fixed

File: src/features/update/edge-morph-animation.ts (hunk 2 above)
Comment (JanKoehnlein) [reply src/features/update/edge-morph-animation.ts]: Added 2020 everywhere

File: src/features/update/update-model.spec.ts (hunk 3 above)
Comment (JanKoehnlein) [reply src/features/update/update-model.spec.ts]: Fixed

File: src/features/move/move.ts (hunk 4 above)
Comment (JanKoehnlein) [reply src/features/move/move.ts]: `EdgeMorphAnimation` has been deleted

**Comment by JanKoehnlein** [comment]
Thanks for the review

**Pull request merged**